### DISCUSSION TITLE
Hot reload?

### DISCUSSION THREAD

**Discussion by tonijota** [sebastienros/jint#1238] [General]
It is possible to do some kind of hot reloading with Jint (3)?

**Comment by lahma** [comment]
Currently I think state cannot be kept between such source code refresh, new Engine instance with new scripts executed into it.

**Comment by nathan130200** [comment]
I have an question related to this also. Its safe multiple instances of new `Engine` for per-script isolated environment?

For example i'm creating an plugin/module based system, that means each script will have two global functions `OnInit` and `OnShutdown` for instanting  and cleanup their data, by script. Currently i'm using per engine instance to do this. 

I'm using same technique like Lua Script State Lifetime, you have per script `lua_State*` then assign all global stuff for each `lua_State*` pointer.

It's ok about memory conditions?

**Reply by lahma** [reply]
Engine instances share very little information, basically just static information that doesn't change, like runtime type information.

Engine instance are also lazy by nature so they shouldn't allocate that many things when created to run simple small scripts.

You probably need to test and benchmark that your expectations are met.

**Reply by nathan130200** [reply]
Thanks!
